src/components/PhoneInput/index.d.ts

```diff
@@ -0,0 +1,35 @@
+import { ReactNode, MouseEvent, FocusEvent, KeyboardEvent } from 'react';
+import { BaseProps } from '../types';
+
+export interface PhoneInputValue {
+    countryCode?: string;
+    isoCode?: string;
+    phone?: string;
+}
+
+export interface PhoneInputProps extends BaseProps {
+    id?: string;
+    name?: string;
+    label?: ReactNode;
+    hideLabel?: boolean;
+    placeholder?: string;
+    icon?: ReactNode;
+    maxLength?: number;
+    minLength?: number;
+    pattern?: string;
```

Comment:
in this case, the pattern prop does not make sense, also let's remove maxLength and minLength for now until a use case appears